### ISSUE TITLE
TypeError: 'DelayedCall' object is not iterable

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8307]
|<img alt="allenap's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| allenap reported|
|-|-|
|Trac ID|trac#8307|
|Type|defect|
|Created|2016-04-26 16:06:50Z|

```
Python 3.5.1+ (default, Mar 30 2016, 22:46:26) 
[GCC 5.3.1 20160330] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from twisted.internet.base import DelayedCall
>>> dc = DelayedCall(1, lambda: None, (), {}, lambda dc: None, lambda dc: None)
>>> dc.debug = True
>>> dc.cancel()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gavin/GitHub/twisted/twisted/internet/base.py", line 94, in cancel
    self._str = bytes(self)
TypeError: 'DelayedCall' object is not iterable
```

```
$ tail -n1 twisted/_version.py 
version = versions.Version('twisted', 16, 1, 1)
```

<details><summary>Searchable metadata</summary>

```
trac-id__8307 8307
type__defect defect
reporter__allenap allenap
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__new new
resolution__None None
component__core core
keywords__None None
time__1461686810312194 1461686810312194
changetime__1462290490013595 1462290490013595
version__None None
owner__None None

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="pawelmhm's avatar" src="https://avatars.githubusercontent.com/u/2700942?s=50" width="50" height="50">](https://github.com/pawelmhm)<a name="note_1"></a>|@pawelmhm commented|
|-|-|

seems like this is caused by [#8110](https://github.com/twisted/twisted/issues/8110) - when I remove dc.debug = True line I dont get any excception

**Comment by exarkun** [comment]
This was fixed by 8f07a5afbbdcd00e387b2d91344f1ac7a1d27354 in #679 for the duplicate issue #8995.
